File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        maxi = SUM = 0
        
        for i in range(len(nums)):
            if i == 0 or nums[i] > nums[i - 1]:
                SUM += nums[i]
            else:
                maxi = max(maxi, SUM)
                SUM = nums[i]
            
        maxi = max(maxi, SUM)
        
        return maxi

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
    maxi, x := nums[0], nums[0]

    for i := 1; i < len(nums); i++ {
        if nums[i] > nums[i-1] {
            x += nums[i]
            if x > maxi {
                maxi = x
            }
        } else {
            x = nums[i]
        }
    }

    return maxi
}